src/DiffOpt.jl

```diff
@@ -1,13 +1,84 @@
 module DiffOpt
 
 using Random
+using LinearAlgebra
 using MathOptInterface
 
 const MOI = MathOptInterface;
 const MOIU = MathOptInterface.Utilities;
 
+include("./gen_random_problem.jl")
 include("./utils.jl")
 
+
+function DiffModel(_model::MOI.AbstractOptimizer, con_idx)
+    
+    model = deepcopy(_model)
+
+    # variable indices
+    var_idx = MOI.get(model, MOI.ListOfVariableIndices())
+    nz      = size(var_idx)[1]
+    
+    # TODO: both Ipopt, OSQP dont support `ListOfConstraintIndices`
+    #       attribute as of now. so passing constraint indices manually
+    eq_con_idx   = MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64},MOI.EqualTo{Float64}}[]
+    ineq_con_idx = MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64},MOI.LessThan{Float64}}[]
+
+    # split into equality and inequality constraints
+    for con in con_idx
+        con_set = MOI.get(model, MOI.ConstraintSet(), con)
+        if is_equality(con_set)
+            push!(eq_con_idx, con)
+        else
+            push!(ineq_con_idx, con)
+        end
+    end
+
+    neq   = size(eq_con_idx)[1]
+    nineq = size(ineq_con_idx)[1]
+    
+    
+    # TODO: fix this method
+    # Q,p,G,h,A,b = generate_matrices(model, ineq_con_idx, eq_con_idx, var_idx)
+    
+    
+    z::Array{Float64} = zeros(0) # solution
```

Comment:
do you need these declarations here?
Also, you don't need the type declaration with it

Reply:
ok will edit them. we need to store the solutions somewhere so that they are accessible to backend method